src/core/dash.h

```diff
@@ -953,7 +958,7 @@ auto DashTable<_Key, _Value, Policy>::GetRandomCursor(absl::BitGen* bitgen) -> C
 
 template <typename _Key, typename _Value, typename Policy>
 template <typename Cb>
-auto DashTable<_Key, _Value, Policy>::Traverse(Cursor curs, Cb&& cb) -> Cursor {
+auto DashTable<_Key, _Value, Policy>::Traverse(Cursor curs, Cb&& cb, YieldCb yield_cb) -> Cursor {
```

Comment:
why is it needed here?
why not yield after the calls to traverse ends?

Reply:
Because `Traverse` will stop once it `finds` and item. If it doesn't find an item, it will `Traverse` over all of the segments/buckets. So we want to avoid that, and that's why I added the YieldCb

Reply:
Oh it uses TraverLogicalBucket() which returns false only if the bucket is empty. +1 I will move the callback out of this :)